digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -0,0 +1,595 @@
+    *                 If multiple means the chain is being updated and only properties are change ( Lane size / cut cannot be done)
+    * @param sideCode To know the direction
+    * @param username Username of the user is doing the changes
+    * @return
+    */
+  def update ( newIncomeLane: Seq[NewIncomeLane], linkIds: Set[Long] ,sideCode: Int, username: String ): Seq[Long] = {
+    if (newIncomeLane.isEmpty || linkIds.isEmpty)
+      return Seq()
+
+    withDynTransaction {
+     if (linkIds.size == 1) {
+        val linkId = linkIds.head
+
+        newIncomeLane.map { lane =>
+
+          val laneCode = getLaneCode(lane)
+
+          val lameToUpdate = PersistedLane(lane.id, linkId, sideCode, laneCode.toInt, lane.municipalityCode,
+            lane.startMeasure, lane.endMeasure, Some(username), None, None, None, false, 0, None, lane.attributes)
+
+          dao.updateEntryLane(lameToUpdate, username)
+        }
+
+      } else {
+
+        // get all lane codes from lanes to update
+        val laneCodes = newIncomeLane.map( lane => getLaneCode(lane).toInt )
+
+        // fetch from db the existing lanes
+        val allExistingLanes = dao.fetchLanesByLinkIdsAndLaneCode(linkIds.toSeq, laneCodes )
+                                  .groupBy(_.linkId)
+
+        // Iterate through the links
+        val result = linkIds.map { linkId =>
+          newIncomeLane.map { lane =>
+
+            val laneCode = getLaneCode(lane).toInt
+
+            val currentLanes = allExistingLanes.getOrElse(linkId, throw new InvalidParameterException(s"LinkId: $linkId dont have laneCode: $laneCode for update!"))
+            val auxLane = currentLanes.filter(_.laneCode == laneCode)
+
+            // Since cut lanes will not be in multiple linkIds Update, we can only have the 1st element
+            val currentLane = if ( auxLane.nonEmpty)
+                                auxLane.head
+                              else
+                                throw new InvalidParameterException(s"LinkId: $linkId dont have laneCode: $laneCode for update!")
+
+
+            val laneToUpdate = PersistedLane(currentLane.id, linkId, sideCode, laneCode, currentLane.municipalityCode,
+              currentLane.startMeasure, currentLane.endMeasure, Some(username), None, None, None, false, 0, None, lane.attributes)
+
+            dao.updateEntryLane(laneToUpdate, username)
+          }
+        }
+
+        result.head
+      }
+    }
+  }
+
+  def updatePersistedLanes ( lanes: Seq[PersistedLane], username: String ): Seq[Long] = {
+    withDynTransaction {
+      lanes.map { lane =>
+        dao.updateEntryLane(lane, username)
+      }
+    }
+  }
+
+  def createWithoutTransaction( newLane: PersistedLane, username: String ): Long = {
+
+    val laneId = dao.createLane( newLane, username )
+    val lanePositionId = dao.createLanePosition(newLane, username)
+
+    dao.createLanePositionRelation(laneId, lanePositionId)
```

Comment:
Unnecessary actions, let's put all inside the createLane method as suggested on LaneDao.scala to keep the same idea of implementation of the old code.

Reply:
Done.